nomad/structs/errors.go

```diff
@@ -13,6 +13,7 @@ const (
 	errNoRegionPath               = "No path to region"
 	errTokenNotFound              = "ACL token not found"
 	errPermissionDenied           = "Permission denied"
+	errJobRegistrationDisabled    = "Job registration, dispatch, and scale are disabled by the scheduler configuration"
```

Comment:
@mikenomitch @davemay99 I updated the error message we emit as discussed, and it'll look like so:

> $ nomad job run ./example.nomad
Error submitting job: Unexpected response code: 403 (Job registration, dispatch, and scale are disabled by the scheduler configuration)

I'd be happy to take any suggestions on improving this message!